src/libraries/System.Linq/tests/ReverseTests.cs

```diff
@@ -48,6 +49,40 @@ public void Reverse<T>(IEnumerable<T> source)
             Assert.Equal(actual, actual); // Repeat the enumeration against itself.
         }
 
+        [Theory]
+        [MemberData(nameof(ReverseData))]
+        public void ReverseArray<T>(IEnumerable<T> source)
+        {
+            T[] expected = source.ToArray();
+            Array.Reverse(expected);
+
+            IEnumerable<T> actual = source.ToArray().Reverse();
```

Comment:
Does this test exercise the case where `source` is empty?

Reply:
Yes, an empty array is in the input data.